DIRSHA256.md

```diff
@@ -0,0 +1,212 @@
+# DIRSHA256 specifications
+
+## Abstract
+
+This document describes DIRSHA256, an algorithm to serialize the content of a filesystem path using the cryptographic hash function SHA256. The cryptographic strength of DIRSHA256 depends on the properties of the SHA256 hash function.
+
+## Introduction
+
+ML models like PyTorch's and Huggingface's store model parameters (weight, architecture, etc) in several files stored in a directory. Models can be several hundreds gigabytes large. Thus we need a fast and flexible serizalization mechanism to cryptographically hash a model stored in a directory.
```

Comment:
Nit: It's Hugging Face (not Huggingface).  Also this comment is referring to how frameworks (not models) save things, right?  So I would say something more like "ML frameworks like PyTorch" (since HF is not a framework anyway).

I would also change the second sentence to "These files can be several hundreds of gigabytes in size"